Project cleanup

jmfee-usgs opened the pull request:
Remove grunt dependency, and just use npm/package.json for task automation.
Preparing to release updated version, and submit a recipe to conda-forge...

erigler-usgs commented:
@jmfee-usgs , 

I'm happy to review the changes, but I suspect that if it is mostly about packaging I won't be able to do much more than nod, and say "yep, looks good to me". Also, can you provide a little more context so I understand the motivation behind all these changes. Thanks.

jmfee-usgs commented:
@erigler-usgs : Take a look at the develop/install docs more carefully than the rest.  As someone who contributes code to this project, I think you can evaluate whether this makes it simpler to get started.

The bulk of the changes remove "grunt" and related plugins that were used to automatically run unit tests and linting.  These aren't typically used in python projects, more typically used in web applications, and weren't needed to run the underlying python tools `nosetests` and `flake8`.  Hopefully this makes it easier for people already familiar with python to get started.

I also cleaned up the develop and install documentation and updated the travis integration configuration to reflect these changes as well as add code unit test coverage.

The last change is to bump the minor release version based on Greg's latest pull request that was merged.
